handlers/ens.go

```diff
@@ -0,0 +1,86 @@
+package handlers
+
+import (
+	"encoding/json"
+	"errors"
+	"eth2-exporter/cache"
+	"eth2-exporter/rpc"
+	"eth2-exporter/types"
+	"eth2-exporter/utils"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/gorilla/mux"
+	ens "github.com/wealdtech/go-ens/v3"
+)
+
+func ResolveEnsDomain(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	vars := mux.Vars(r)
+	search := vars["domain"]
+
+	data, err := GetEnsDomain(search)
+
+	if err != nil {
+		logger.Warnf("failed to resolve ens %v: %v", search, err)
+		sendErrorResponse(w, r.URL.String(), "failed to resolve ens")
+		return
+	}
+
+	j := json.NewEncoder(w)
+	sendOKResponse(j, r.URL.String(), []interface{}{data})
+}
+
+func GetEnsDomain(search string) (*types.EnsDomainResponse, error) {
+	data := &types.EnsDomainResponse{}
+	var returnError error
+
+	if utils.IsValidEnsDomain(search) {
+		data.Domain = search
+		cacheKey := fmt.Sprintf("%d:ens:address:%v", utils.Config.Chain.Config.DepositChainID, search)
+
+		if address, err := cache.TieredCache.GetStringWithLocalTimeout(cacheKey, time.Hour); err != nil || len(address) == 0 {
```

Comment:
As discussed with @peterbitfly, caching introduces inaccuracy as ENS names can change every block.
Removing caching with the implementation here would potentially mean a lot of traffic for our node.

I am afraid we cannot implement this whole feature the way you did. @peterbitfly suggested to index events on the according ENS contract in bigtable to always be up2date and to reduce the load on our nodes.